VideoWeb/VideoWeb/ClientApp/src/app/services/video-filter.service.ts

```diff
@@ -0,0 +1,172 @@
+import { Injectable } from '@angular/core';
+import { Camera } from '@mediapipe/camera_utils';
+import { Results, SelfieSegmentation } from '@mediapipe/selfie_segmentation';
+import { Observable, Subject } from 'rxjs';
+import { Logger } from './logging/logger-base';
+import { BackgroundFilter } from './models/background-filter';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class VideoFilterService {
+    private readonly loggerPrefix = '[VideoFilterService] -';
+
+    private _onFilterChanged = new Subject<BackgroundFilter | null>();
+    get onFilterChanged$(): Observable<BackgroundFilter | null> {
+        return this._onFilterChanged.asObservable();
+    }
+
+    videoElement: HTMLVideoElement;
+    canvasElement: HTMLCanvasElement;
+    canvasStream: MediaStream;
+
+    canvasCtx: CanvasRenderingContext2D;
+
+    filterOn: boolean;
+    selfieSegmentation: SelfieSegmentation;
+    activeFilter: BackgroundFilter;
+    imgs: Map<BackgroundFilter, HTMLImageElement> = new Map();
+
+    constructor(private logger: Logger) {
+        this.filterOn = false;
+        this.activeFilter = null;
+
+        this.selfieSegmentation = new SelfieSegmentation({
+            locateFile: file => {
+                return `https://cdn.jsdelivr.net/npm/@mediapipe/selfie_segmentation@0.1.1628007100/${file}`;
```

Comment:
Any way we could change this to something we have control over?

Reply:
What do you mean? Specifying the version number is total control. Also, this number should change in line with the version in the package.json file